validators/validators.go

```diff
@@ -20,8 +21,12 @@ var (
 	ErrInvalidTimestamp = errors.New("invalid timestamp")
 )
 
+type MultiSlotState interface {
+	Get(uint64) State
+}
+
 type State interface {
-	Duties() structs.DutiesState
+	Duties(uint64) structs.DutiesState
```

Comment:
```suggestion
	Duties(uint64) structs.DutiesState
	HeadSlot() uint64
```
? :)

Reply:
I tried to do this, but it becomes an infinite rabbit hole, so I will open a consequent PR for getting rid of `structs.Slot` and `structs.Epoch` entirely